panels/user-accounts/cc-avatar-chooser.c

```diff
@@ -411,77 +411,133 @@ cheese_camera_device_monitor_new_cb (GObject *source,
+                g_ptr_array_add (facesdirs, path);
+        }
+
+        return (GStrv) g_steal_pointer (&facesdirs->pdata);
+}
+
+static gboolean
+add_faces_from_dirs (GListStore *faces, GStrv facesdirs, gboolean add_all)
+{
+        gboolean added_faces = FALSE;
+        const gchar *target;
+        int i;
+        GFileType type;
 
+        for (i = 0; facesdirs[i] != NULL; i++) {
+                g_autoptr(GFileEnumerator) enumerator = NULL;
+                g_autoptr(GFile) dir = NULL;
+                const char *path = facesdirs[i];
+                gpointer infoptr;
+
+                dir = g_file_new_for_path (path);
                 enumerator = g_file_enumerate_children (dir,
                                                         G_FILE_ATTRIBUTE_STANDARD_NAME ","
                                                         G_FILE_ATTRIBUTE_STANDARD_TYPE ","
                                                         G_FILE_ATTRIBUTE_STANDARD_IS_SYMLINK ","
                                                         G_FILE_ATTRIBUTE_STANDARD_SYMLINK_TARGET,
                                                         G_FILE_QUERY_INFO_NONE,
                                                         NULL, NULL);
-                if (enumerator == NULL) {
-                        g_object_unref (dir);
+
+                if (enumerator == NULL)
                         continue;
-                }
 
-                while ((info = g_file_enumerator_next_file (enumerator, NULL, NULL)) != NULL) {
-                        added_faces = TRUE;
+                while ((infoptr = g_file_enumerator_next_file (enumerator, NULL, NULL)) != NULL) {
+                        g_autoptr (GFileInfo) info = infoptr;
+                        g_autoptr (GFile) face_file = NULL;
 
                         type = g_file_info_get_file_type (info);
-                        if (type != G_FILE_TYPE_REGULAR &&
-                            type != G_FILE_TYPE_SYMBOLIC_LINK) {
-                                g_object_unref (info);
+                        if (type != G_FILE_TYPE_REGULAR && type != G_FILE_TYPE_SYMBOLIC_LINK)
                                 continue;
-                        }
 
                         target = g_file_info_get_symlink_target (info);
-                        if (target != NULL && g_str_has_prefix (target , "legacy/")) {
-                                g_object_unref (info);
+                        if (target != NULL && g_str_has_prefix (target , "legacy/"))
                                 continue;
-                        }
 
-                        file = g_file_get_child (dir, g_file_info_get_name (info));
-                        g_list_store_append (self->faces, file);
-
-                        g_object_unref (info);
+                        face_file = g_file_get_child (dir, g_file_info_get_name (info));
+                        g_list_store_append (faces, face_file);
```

Comment:
Don't we need to use g_steal_pointer here for face_file? Or not use g_autoptr for it.

Reply:
Clarified upstream, g_list_store_append takes a ref of the arg.